tests/conftest.py

```diff
@@ -15,6 +15,17 @@
 
 from uvicorn.config import LOGGING_CONFIG
 
+# Note: We explicitly turn the propagate on just for tests, because pytest
+# caplog not able to capture no-propagate loggers.
+#
+# And the caplog_for_logger helper also not work on test config cases, because
+# when create Config object, Config.configure_logging will remove caplog.handler.
+#
+# The simple solution is set propagate=True before execute tests.
+#
+# See also: https://github.com/pytest-dev/pytest/issues/3697
+LOGGING_CONFIG["loggers"]["uvicorn"]["propagate"] = True
```

Comment:
But then... all our users needs to do this if capturing the logs on their setup, right? 🤔 

Should we mention this somewhere, or it's fine?

Reply:
> all our users needs to do this if capturing the logs on their setup, right?

I think if users want to config uvicorn's logs, they should provide full logging config, no needs to set propagate. https://www.uvicorn.org/settings/#logging

And if users only want to config application logs, call `logging.basicConfig` works as expected, uvicorn's logs will not affected.

Reply:
I mean, if people want to capture those logs on their test, for some reason (?) they need to do the same thing.

Reply:
> But then... all our users needs to do this if capturing the logs on their setup, right? thinking
> 
> Should we mention this somewhere, or it's fine?

It introduces a change only for the users who in their tests, use a Config object and are looking for logs if I get it correctly, logging is complex so I'll admit I'm not 100% sure.

I'd be tempted to say it's a rather slim part of people in that case but I may be proven wrong, but there's no harm in saying something like:

> Should you use the caplog pytest fixture in your test suite to capture uvicorn logs, you might want to use (the line above) if you use a Config object, please see (test line that explains it)

Reply:
thoughts on this @guyskk ?

Reply:
(I just wanted to point this out, I don't have any strong opinion about this, jfyk)

Reply:
> I mean, if people want to capture those logs on their test, for some reason (?) they need to do the same thing.

Yes. Maybe mention it in changelog? if those user's tests broken(rare usage), they can easily know why.

Reply:
ok let's add a note then !